omegaconf/listconfig.py

```diff
@@ -527,10 +527,7 @@ def _set_value_impl(
             self.__dict__["_content"] = None
         elif get_value_kind(value) == ValueKind.MANDATORY_MISSING:
             self.__dict__["_content"] = "???"
-        elif get_value_kind(value) in (
-            ValueKind.INTERPOLATION,
-            ValueKind.STR_INTERPOLATION,
-        ):
+        elif get_value_kind(value) == ValueKind.INTERPOLATION:
```

Comment:
Can you explain this change? I think we discussed it but I can't remember the details.

Reply:
I got rid of `ValueKind.STR_INTERPOLATION` because I didn't see any reason to make a difference with other kinds of interpolations.